docs/self-managed/zeebe-deployment/security/secure-cluster-communication.md

```diff
@@ -0,0 +1,174 @@
+    enabled: false
+
+    # Sets the path to the certificate chain file.
+    # This setting can also be overridden using the environment variable ZEEBE_GATEWAY_CLUSTER_SECURITY_CERTIFICATECHAINPATH.
+    certificateChainPath:
+
+    # Sets the path to the private key file location
+    # This setting can also be overridden using the environment variable ZEEBE_GATEWAY_CLUSTER_SECURITY_PRIVATEKEYPATH.
+    privateKeyPath:
+```
+
+:::note
+
+The `certificateChainPath` and the `privateKeyPath` can be relative to the gateway's working directory, or can be absolute paths.
+
+:::
+
+## How it works
+
+When enabled for each node, communication over TCP between these will be securely encrypted using the provided certificates, in a client-server model. So given two nodes, `A` and `B`, when `A` (the client) sends a request to `B` (the server), they will perform a TLS handshake wherein `B`'s certificate will be exchanged and verified by `A`, after which the request will be encrypted such that only a node with `B`'s private key may decrypt it (i.e. in this instance, `B`).
+
+When the roles are reversed (e.g. `B` sends a request to `A`), the same handshake will occur but the other way around. As`B` is now the client, and `A` the server, `A`'s certificate is exchanged and verified by `B`, after which all communication is encrypted and can only be decrypted with `A`'s private key.
+
+:::note
+
+In this model, only the client verifies the identity of the server, as opposed to mTLS, in which both client and server would exchange and verify each other's identities. If you need mTLS, it's currently recommended that you explore a solution which provides this transparently like a service mesh (e.g. Linkerd or Istio).
+
+:::
+
+## Self signed certificates
+
+If you wish to use self signed certificates for testing or development purposes, the simplest way is to have all nodes share the same certificate. As aforementioned, the certificate chain configured on a node is also part of its trust store. As such, if all nodes share the same certificate, they will have no trouble verifying the identity of the other nodes.
+
+Of course, you can still configure a different self signed certificate for each node, _provided that they can be verified by the other nodes' certificate chain_. For example, let's say you have your own root certificate authority you use to sign your own certificates, and one certificate for each node that you signed with that authority. For each node, you can then create a certificate chain file which would consist of the node's public certificate, followed by the root certificate authority's public certificate. Even though each node would have a different leaf certificate it uses to identify itself, the other nodes could verify its identity since their certificate chain contains an authority used to sign it.
+
+### Testing & example
+
+To generate your own self signed certificates for testing, you must first create a certificate authority.
```

Comment:
Awesome I wanted to ask for an example or link earlier, nice you have added this. :+1: